src/lib/functions/server.js

```diff
@@ -45,7 +47,49 @@ const buildClientContext = function (headers) {
   }
 }
 
-const createHandler = function ({ config, functionsRegistry }) {
+const startPollingForAPIAuthentication = async function (options) {
+  const { api, command, config, site, siteInfo } = options
+  const frequency = 5000
+
+  const helper = async (maybeSiteData) => {
+    const siteData = await (maybeSiteData || api.getSite({ siteId: site.id }))
+    const authlifyTokenId = siteData && siteData.authlify_token_id
+
+    const existingAuthlifyTokenId = config && config.netlifyGraphConfig && config.netlifyGraphConfig.authlifyTokenId
+    if (authlifyTokenId && authlifyTokenId !== existingAuthlifyTokenId) {
+      const netlifyToken = await command.authenticate()
+      // Only inject the authlify config if a token ID exists. This prevents
+      // calling command.authenticate() (which opens a browser window) if the
+      // user hasn't enabled API Authentication
+      const netlifyGraphConfig = {
+        netlifyToken,
+        authlifyTokenId: siteData.authlify_token_id,
+        siteId: site.id,
+      }
+      config.netlifyGraphConfig = netlifyGraphConfig
+
+      const netlifyGraphJWT = generateNetlifyGraphJWT(netlifyGraphConfig)
+
+      if (netlifyGraphJWT != null) {
+        process.env.ONEGRAPH_AUTHLIFY_TOKEN = netlifyGraphJWT
+      }
+    } else {
+      delete config.authlify
+    }
+
+    setTimeout(helper, frequency)
```

Comment:
Have we considered using https://github.com/sindresorhus/p-wait-for#p-wait-for to abstract the polling part?

Reply:
This sounds like a good improvement to make as a follow-up PR. I'm opening an issue to change our pollers in Graph to use `p-wait-for`.

Reply:
https://github.com/netlify/cli/issues/4256